File: tests/nft/src/test/java/com/example/UrlShortenerSimulation.java
package com.example;

import io.gatling.javaapi.core.*;
import io.gatling.javaapi.http.*;

import static io.gatling.javaapi.core.CoreDsl.*;
import static io.gatling.javaapi.http.HttpDsl.*;

import java.time.Duration;

public class UrlShortenerSimulation extends Simulation {

    HttpProtocolBuilder httpProtocol = http
            .baseUrl("http://localhost:8080")
            .acceptHeader("application/json");

    ScenarioBuilder scn = scenario("Shorten URL")
            .exec(http("Shorten URL")
                    .post("/shorten")
                    .body(StringBody(session -> "{\"fullUrl\": \"https://example.com/very/long/url\", \"customAlias\": \"test\" + java.util.UUID.randomUUID().toString() + \"}"))
                    .asJson()
                    .check(status().is(201)))
            .pause(Duration.ofSeconds(1))
            .exec(http("Redirect")
                    .get(session -> "/test" + java.util.UUID.randomUUID().toString())
                    .check(status().in(302, 404)));

    {
        setUp(
                scn.injectOpen(
                        constantUsersPerSec(10).during(Duration.ofSeconds(30))
                )
        ).protocols(httpProtocol);
    }
}

File: tests/nft/src/test/scala/com/example/UrlShortenerSimulation.scala
package com.example

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import scala.concurrent.duration._

class UrlShortenerSimulation extends Simulation {
  val httpProtocol = http
    .baseUrl("http://localhost:8080")
    .acceptHeader("application/json")

  val scn = scenario("Shorten URL")
    .exec(http("Shorten URL")
      .post("/shorten")
      .body(StringBody("""{"fullUrl": "https://example.com/very/long/url", "customAlias": "test${randomUUID()}"}""")).asJson
      .check(status.is(201)))
    .pause(1)
    .exec(http("Redirect")
      .get("/test${randomUUID()}")
      .check(status.in(302, 404)))

  setUp(
    scn.inject(
      constantUsersPerSec(10) during (30 seconds)
    )
  ).protocols(httpProtocol)
}
